packages/darklang/languageTools/parser/typeReference.dark

```diff
@@ -0,0 +1,203 @@
+module Darklang =
+  module LanguageTools =
+    module Parser =
+      module TypeReference =
+        let parseBuiltIn
+          (node: ParsedNode)
+          : Stdlib.Result.Result<WrittenTypes.TypeReference.Builtin, WrittenTypes.Unparseable> =
+          if node.typ == "builtin_type" then
+            match node.text with
+            | "Unit" ->
```

Comment:
Not too important, but if you
- created an alias `type TBuiltin = WrittenTypes.TypeReference.Builtin`
- created a small helper fn `let builtin (b: TBuiltin) = Stdlib.Result.Result.Ok b`
, then this could be a bit tidier, just a bunch of single lines like
`| "Unit"  -> TBuiltin.TUnit node.sourceRange`